test/common/filter/tcp_proxy_test.cc

```diff
@@ -47,13 +74,235 @@ TEST(TcpProxyConfigTest, BadTcpProxyConfig) {
       Json::Exception);
 }
 
+TEST(TcpProxyConfigTest, Routes) {
+  std::string json = R"EOF(
+    {
+      "stat_prefix": "name",
+      "route_config": {
+        "routes": [
+          {
+            "destination_ip_list": [
+              "10.10.10.10/32",
+              "10.10.11.0/24",
+              "10.11.0.0/16",
+              "11.0.0.0/8",
+              "128.0.0.0/1"
+            ],
+            "cluster": "with_destination_ip_list"
+          },
+          {
+            "destination_ports": "1-1024,2048-4096,12345", 
+            "cluster": "with_destination_ports"
+          },
+          {  
+            "source_ports": "23457,23459", 
+            "cluster": "with_source_ports"
+          },
+          {
+            "destination_ip_list": [
+              "10.0.0.0/24"
+            ],
+            "source_ip_list": [
+              "20.0.0.0/24"
+            ],
+            "destination_ports" : "10000",
+            "source_ports": "20000",
+            "cluster": "with_everything"
+          },
+          {
+            "cluster": "catch_all"
+          }
+        ]
+      }
+    }
+    )EOF";
+
+  Json::ObjectPtr json_config = Json::Factory::LoadFromString(json);
+  NiceMock<Upstream::MockClusterManager> cm_;
+
+  // The TcpProxyConfig constructor checks if the clusters mentioned in the route_config are valid.
+  // We need to make sure to return a non-null pointer for each, otherwise the constructor will
+  // throw an exception and fail.
+  EXPECT_CALL(cm_, get("with_destination_ip_list")).WillRepeatedly(Return(cm_.cluster_.info_));
```

Comment:
CM mock already has:
ON_CALL(*this, get(_)).WillByDefault(Return(cluster_.info_));

So I don't think any of these lines are necessary?